app/src/main/java/com/mashkovska/authentication/SignUpActivity.java

```diff
@@ -0,0 +1,110 @@
+import android.util.Patterns;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+public class SignUpActivity extends AppCompatActivity {
+    private EditText emailId;
+    private EditText password;
+    private EditText name;
+    private EditText phoneNumber;
+    private Button signUp;
+    private TextView signIn;
+    private FirebaseAuth mFirebaseAuth;
+    private static Integer minLength = 6;
+    private boolean successValidation = true;
+
+    @SuppressLint("WrongConstant")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_sign_up);
+
+        mFirebaseAuth = FirebaseAuth.getInstance();
+
+        emailId = findViewById(R.id.sign_up_screen_email);
+        password = findViewById(R.id.sign_up_screen_password);
+        name = findViewById(R.id.sign_up_screen_name);
+        phoneNumber = findViewById(R.id.sign_up_screen_phone);
+        signUp = findViewById(R.id.sign_up_screen_button);
+        signIn = findViewById(R.id.sign_up_screen_login);
+
+        signUp.setOnClickListener(view -> {
+            final String yourName = name.getText().toString().trim();
+            final String phone = phoneNumber.getText().toString().trim();
+            final String email = emailId.getText().toString().trim();
+            final String pwd = password.getText().toString().trim();
+
+            if (Validation(yourName, phone, email, pwd)) {
+                mFirebaseAuth.createUserWithEmailAndPassword(email, pwd)
+                        .addOnCompleteListener(SignUpActivity.this, task -> {
+                            if (task.isSuccessful()) {
+                                Intent intToHome = new Intent(this,
+                                        WelcomeActivity.class);
+                                intToHome.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP
+                                        | Intent.FLAG_ACTIVITY_SINGLE_TOP
+                                        | Intent.FLAG_ACTIVITY_NO_HISTORY);
+                                startActivity(intToHome);
+                            } else {
+                                Toast.makeText(SignUpActivity.this, "SignUp Unsuccessful",
+                                        Toast.LENGTH_SHORT).show();
+                            }
+                        });
+            }
+        });
+
+        signIn.setOnClickListener(view -> {
+            Intent i = new Intent(SignUpActivity.this, LoginActivity.class);
+            i.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP
+                    | Intent.FLAG_ACTIVITY_NO_HISTORY);
+            startActivity(i);
+        });
+
+
+    }
+
+    public boolean Validation(String yourName, String phone, String email, String pwd) {
```

Comment:
What will be if not only the name field will be empty? Will the user be informed that other fields are empty too? In my opinion, there are too many unnecessary "else" blocks.